server/src/main/java/com/cloud/ha/HighAvailabilityManagerImpl.java

```diff
@@ -795,7 +820,7 @@ private long getRescheduleTime(WorkType workType) {
             case ForceStop:
                 return ((System.currentTimeMillis() >> 10) + _stopRetryInterval);
             case Destroy:
-                return ((System.currentTimeMillis() >> 10) + _restartRetryInterval);
+                return ((System.currentTimeMillis() >> 10) + _stopRetryInterval);
```

Comment:
@shwstppr any reason to change to stop retry interval? if you want to keep it, move the case stmt after 'ForceStop'.

Alternatively, you can use different config (may be "_destroy.retry.interval_") for destroy operation, and set its value to the current value in "_restart.retry.interval_" on upgrade.

Reply:
@sureshanaparti clubbed `Destroy` case with other cases.
Used `_stopRetryInterval` because it uses value of config - `stop.retry.interval`. The description for that config is - `The time in seconds between retries to stop or destroy a VM.`

https://github.com/apache/cloudstack/blob/4.15/engine/components-api/src/main/java/com/cloud/ha/HighAvailabilityManager.java#L50-L51